core/storage/certificate_assessment/gae_models.py

```
@@ -274,3 +276,265 @@ def create(
+        for _ in range(base_models.MAX_RETRIES):
+            new_id = utils.convert_to_hash(
+                str(utils.get_random_int(base_models.RAND_RANGE)),
+                base_models.ID_LENGTH,
+            )
+            if not cls.get_by_id(new_id):
+                return new_id
+
+        raise Exception(
+            'The id generator for CertificateAssessmentAttemptModel '
+            'is producing too many collisions.'
+        )
```

Ruff (0.15.20)

[warning] Create your own exception

(TRY002)